docs/reference/esql/functions/values.asciidoc

```diff
@@ -0,0 +1,38 @@
+[discrete]
+[[esql-agg-values]]
+=== `VALUES`
+
+preview::["Do not use `VALUES` on production environments. This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+
+*Syntax*
+
+[source,esql]
+----
+VALUES(expression)
+----
+
+`expression`::
+Expression of any type except `geo_point`, `cartesian_point`, or `geo_shape`.
```

Comment:
We should gracefully reject the queries using these data types. Atm the error message is pretty ugly (`Cannot find intermediate state for: AggDef[aggClazz=class org.elasticsearch.xpack.esql.expression.function.aggregate.Values, type=GeoPoint, extra=, grouping=false]`).
This should be handled in `Verifier` very similarly to `checkForSortOnSpatialTypes`.

Reply:
That `intermediate state` error message is pretty terrible. We already do type checking in the function itself so I'm wondering if we can't let that go first. Will have a look.

Reply:
It turns out that I wasn't properly type checking in the function class itself. Once I fixed that the error message got a lot better.